src/game/objects/har.c

```diff
@@ -649,6 +649,35 @@ void har_move(object *obj) {
         } else if(h->state == STATE_WALKFROM) {
             obj->pos.x -= (h->back_speed * object_get_direction(obj)) * (h->hard_close ? 0.5 : 1.0);
         }
+
+        // TODO unify this code with the (very similar) code in har_move
+        // but we currently don't have a way in player.c to know if an object is a HAR or a projectile
+        if(player_frame_isset(obj, "cx")) {
+            float cx = player_frame_get(obj, "cx") / 10.0 * obj->horizontal_velocity_modifier;
+            log_debug("CX! %f, %c", cx, h->inputs[0]);
+            if(h->inputs[0] == '4') {
```

Comment:
Switch-case in a shared function between har and projectile ?

Reply:
Possibly a helper function in object.c ?

Reply:
Refactored into a common function as requested.